src/Dispatcher.cpp

```diff
@@ -83,7 +88,7 @@ void DispatcherBase::StartServer(
     if (m_active) return;
     m_active = true;
   }
-  m_server = true;
+  m_networkMode = NT_NET_MODE_SERVER;
```

Comment:
Might be worthwhile to have `NT_NET_MODE_SERVER_FAIL` and `NT_NET_MODE_CLIENT_FAIL` as well instead of having `NT_NET_MODE_NONE` be a catchall